src/crimson/osd/osd.h

```diff
@@ -87,7 +87,10 @@ class OSD : public ceph::net::Dispatcher,
   seastar::future<> mkfs(uuid_d fsid);
 
   seastar::future<> start();
-  seastar::future<> stop();
+  seastar::future<> stop(){
+    return seastar::make_ready_future<>();
```

Comment:
i think we should make proper use of `stop()`. could you elaborate a little bit here?

Reply:
here I try to decouple messenger shard and osd shard stop, messenger shard is execute first, then cause call virtual function error.

Reply:
call pure virtual function error since derived class obj is freed before we call heartbeat::shutdown.

Reply:
and with --mkfs, can't stop all messengers since they are only initialized in OSD::start, but in --mkfs case, OSD::start is not executed, still need a  OSD shard stop function.